Subject: [PATCH 09/21] Exclude overlays from the app counter

This exclude overlays from the app counter that shows
in manage app summary.

diff --git a/src/com/android/settings/applications/AppCounter.java b/src/com/android/settings/applications/AppCounter.java
index fb8d5809e7..852358dca3 100644
--- a/src/com/android/settings/applications/AppCounter.java
+++ b/src/com/android/settings/applications/AppCounter.java
@@ -47,6 +47,7 @@ public abstract class AppCounter extends AsyncTask<Void, Void, Integer> {
                 ParceledListSlice<ApplicationInfo> list =
                         mIpm.getInstalledApplications(PackageManager.GET_DISABLED_COMPONENTS
                                 | PackageManager.GET_DISABLED_UNTIL_USED_COMPONENTS
+                                | PackageManager.GET_META_DATA
                                 | (user.isAdmin() ? PackageManager.GET_UNINSTALLED_PACKAGES : 0),
                                 user.id);
                 for (ApplicationInfo info : list.getList()) {
diff --git a/src/com/android/settings/applications/ManageApplications.java b/src/com/android/settings/applications/ManageApplications.java
index 5d3b353766..ed617814b3 100644
--- a/src/com/android/settings/applications/ManageApplications.java
+++ b/src/com/android/settings/applications/ManageApplications.java
@@ -1295,6 +1295,12 @@ public class ManageApplications extends InstrumentedFragment
                         if ((info.flags & ApplicationInfo.FLAG_UPDATED_SYSTEM_APP) != 0) {
                             return true;
                         } else if ((info.flags & ApplicationInfo.FLAG_SYSTEM) == 0) {
+                            if (info.metaData != null) {
+                                if (info.metaData.getString("Substratum_Parent") != null
+                                        || info.metaData.getString("Substratum_IconPack") != null) {
+                                    return false;
+                                }
+                            }
                             return true;
                         }
                         Intent launchIntent = new Intent(Intent.ACTION_MAIN, null)
